tests/visualisations/test_trackmap.py

```diff
@@ -19,16 +28,109 @@ def prefilled_trackmap(example_plottingdataframe):
+
+
+@pytest.fixture
+def lines_geodataframe():
+    line1 = LineString([Point(4.9, 52.1), Point(4.95, 51.1)]).wkt
+    line2 = LineString([Point(5.0, 52.1), Point(4.95, 51.1)]).wkt
+    line3 = LineString([Point(4.8, 52.1), Point(4.95, 51.1)]).wkt
+
+    return (
+        pd.DataFrame(data={'name': ['Q', 'R', 'S'],
+                           'name2': ['QQ', 'RR', 'SS'],
+                           'geometry': [line1, line2, line3]})
+        .assign(geometry=lambda d: d.geometry.apply(wkt.loads))
+        .pipe(gpd.GeoDataFrame, geometry='geometry', crs='EPSG:4326')
+        .assign(geometry=lambda d: d.geometry.to_crs('EPSG:4326'))
+    )
+
+
+@pytest.fixture
+def areas_geodataframe():
+    area1 = Polygon([Point(5.0, 52.0), Point(5.1, 52.0), Point(5.1, 52.1), Point(5.0, 52.1)]).wkt
+    area2 = Polygon([Point(5.5, 52.0), Point(5.6, 52.0), Point(5.6, 52.1), Point(5.5, 52.1)]).wkt
+
+    return (
+        pd.DataFrame(data={'name': ['M', 'N'], 'name2': ['MM', 'NN'], 'geometry': [area1, area2]})
+        .assign(geometry=lambda d: d.geometry.apply(wkt.loads))
+        .pipe(gpd.GeoDataFrame, geometry='geometry', crs='EPSG:4326')
+        .assign(geometry=lambda d: d.geometry.to_crs('EPSG:4326'))
+    )
+
+
+def test_add_to_trackmap(tmp_path, points_dataframe, lines_geodataframe, areas_geodataframe):
+    m = TrackMap()
+
+    PlottingPoints(points_dataframe, popup=['name2']).add_to(m)
+    PlottingLineStrings(lines_geodataframe, popup=['name2']).add_to(m)
+    PlottingAreas(areas_geodataframe, popup=['name2']).add_to(m)
+    total_objects = len(points_dataframe) + len(lines_geodataframe) + len(areas_geodataframe)
+
+    assert len(m._children) == total_objects, "Invalid number of items added"
+
+    m.save(tmp_path / 'test_output.html')
+    # One more bounds fitted object
+    assert len(m._children) == 1 + total_objects, "Invalid number of items added"
+
+    assert 'test_output.html' in os.listdir(tmp_path), 'No file was created'
+
+    # Make a new map, that should be inferred to show the same data
+    # Type is established with plottable, and popup is a string instead of a list
+    q = TrackMap()
+    plottable(points_dataframe, popup='name').add_to(q)
+    plottable(lines_geodataframe, popup='name').add_to(q)
+    plottable(areas_geodataframe, popup='name').add_to(q)
+    q.save(tmp_path / 'test_output2.html')
+
+    for m_child, q_child in zip(m._children, q._children):
+        assert type(m_child) == type(q_child), 'Unequal type'
+
+    # Popup is a list with 2 elements, and test some additional settings for linestrings and areas
+    r = TrackMap()
+    PlottingPoints(points_dataframe, popup=['name', 'name2']).add_to(r)
+    PlottingLineStrings(lines_geodataframe, popup=['name', 'name2'], color='blue').add_to(r)
+    PlottingAreas(areas_geodataframe, popup=['name', 'name2'], color='orange').add_to(r)
+    r.save(tmp_path / 'test_output3.html')
+
+    for q_child, r_child in zip(q._children, r._children):
```

Comment:
For consistency reasons I would zip m._children and r._children instead of q._children and m._children. This to avoid any future testing problems when changing for instance plottable.

Reply:
Agreed, will fix!